### PR TITLE
Feature/ge user creation

### LINKED ISSUE #63: User CREATION

### PR DESCRIPTION
**Pull request by GreysonElkins**
#### What's this PR do?
* Login component can now handle signing up a user as well!
* Upon a user's creation, they'll automatically be logged in
* Some error handling refactorization
#### Where should the reviewer start?
`Login` component
`verifySignup` vs. `verifyLogin` probably have plenty of oppurtunity for refactor and combination
#### How should this be manually tested?
Make some users!
#### Any background context you want to provide?
The error handling isn't quite as perfect as it should be; I don't know how to access the errors you'd usually get in PostMan if you mess stuff up, but they'd be helpful.
#### What are the relevant tickets?
close #63 
#### Questions:
Anyone know how to access the error from a fetch response? The BE responds as such: 
`response.status(422).json('You seriously messed up Mr. Elkins')`

### CODE REVIEW

File: src/components/Login/Login.js [hunk 1]
```diff
@@ -51,6 +53,73 @@ class Login extends Component {
...
+      this.state.username, 
+      this.state.email
+    ]
+
+    if(userCredentials.some(cred => cred === '')){
+      this.setState({ submitEmpty: true })
+    } else if (userCredentials[0] !== userCredentials[1]) {
+      this.setState({ error: 'bad passwords'})
+    } else if (!userCredentials[3].includes('@')) {
+      this.setState({ error: "that's not an email"})
```
Comment (nickhartdev) [inline src/components/Login/Login.js:71]: I thought the 422 response code from the server had a message attached to it? It seems redundant to have all the error handling on the server side and on the front end

File: src/components/Login/Login.js [hunk 2]
```diff
@@ -51,6 +53,73 @@ class Login extends Component {
...
+    let error = this.state.error
+    if (this.state.submitEmpty === true) {
+      warning = `One or more fields are empty`
+    } else if (error === "Incorrect login credentials") {
+      warning = `We were unable to find a user with those credentials,`
+    } else if (error === "bad passwords") {
+      warning = `Those passwords don't match!`
+    } else if (error === "server") {
+      warning = 'Something went wrong,'
+    } else if (error === "That's not an email,") {
```
Comment (nickhartdev) [inline src/components/Login/Login.js:105]: Same idea with these conditionals

**Review by nickhartdev — APPROVED** [review]
Looks good! Only piece of feedback is the slight redundancy with the error handling - I thought you could just use `catch` to return the messages that were meant to be sent back from the server. I appreciate how thorough you've been with it, but I think it only really needs to happen on one side of the stack or the other

Edit: After reading the last question, wouldn't it just be accessed by the catch? Specifically in `error.message`

File: src/components/App/App.js [hunk 3]
```diff
@@ -207,6 +207,15 @@ class App extends Component {
             /> 
           }}
         /> 
+        <Route 
+          exact path='/sign-up'
+          render={() => {
```
Comment (Abdeboskey) [inline src/components/App/App.js:212]: I wonder if a pathname `/create-user` would be more specific here?

File: src/components/Login/Login.js [hunk 4]
```diff
@@ -10,7 +10,9 @@ class Login extends Component {
       this.state = {
         username: '',
         password: '',
-        submitEmptyLogin: false,
+        verifyPassword: '',
+        email: '',
+        submitEmpty: false,
         redirect: false,
         error: ''
```
Comment (Abdeboskey) [inline src/components/Login/Login.js:17]: I think we could bundle the user info into an object in state here. I got remarks from instructors about that on the last proj. 
`this.state = {
  user: { username: '', password: '', email: ''}
...`
just a thought! maybe a better issue for a later time.

File: src/components/Login/Login.js (hunk 1 above)
Comment (Abdeboskey) [reply src/components/Login/Login.js:71]: Yes, I thought it was just `error.message`

**Review by Abdeboskey — APPROVED** [review]
Looks good bubs

**Review by carlymclift — APPROVED** [review]
This will be a great addition!

**Pull request merged**